lib/core/elastic_connector_actions.rb

```diff
@@ -88,13 +88,19 @@ def claim_job(connector_package_id)
         job['_id']
       end
 
-      def heartbeat(connector_package_id)
+      def heartbeat(connector_package_id, service_type)
         body = {
           :doc => {
             :last_seen => Time.now
           }
         }
 
+        connector_setting = Core::ConnectorSettings.fetch(connector_package_id)
+        if connector_setting.connector_status_allows_sync?
+          connector_instance = Connectors::REGISTRY.connector(service_type, connector_setting.configuration)
+          body[:doc][:status] = connector_instance.source_status[:status] == 'OK' ? Connectors::ConnectorStatus::CONNECTED : Connectors::ConnectorStatus::ERROR
+        end
+
```

Comment:
ElasticConnectorActions method is not supposed to have any domain logic - it's only a wrapper around calls to Elasticsearch that are needed to support BYOC and BYOEI protocols.

I think this code should be pulled up to the caller of the method or into another class that encapsulates the logic.

Reply:
OK, will look deeper tomorrow